opm/output/eclipse/AggregateWellData.cpp

```diff
@@ -502,6 +532,80 @@ namespace {
                 }
             }
+        }
+
+        template <typename IWellArray>
+        void staticContribWellHeadLGR(const        Opm::Well&    well,
+                                                 IWellArray&    iWell,
+                                     const Opm::EclipseGrid&    grid)
+        {
+            using Ix = VI::IWell::index;
+
+            const std::size_t I = well.getHeadI();
+            const std::size_t J = well.getHeadJ();
+
+            // Connections for LGR well
+            {
+                std::size_t firstK, lastK;
+                const auto& conn = well.getConnections();
+
+                iWell[Ix::NConn]  = static_cast<int>(conn.size());
+
+                if (well.isMultiSegment()) {
+                    // Set top and bottom connections to zero for multi
+                    // segment wells
+                    firstK = 0;
+                    lastK  = 0;
+                }
+                else {
+                    firstK = (iWell[Ix::NConn] == 0)
+                        ? 0 : conn.get(0).getK();
+
+                    lastK=  (iWell[Ix::NConn] == 0)
+                        ? 0 : conn.get(conn.size() - 1).getK();
+                }
+                auto lowerIJK = grid.getLGR_fatherIJK(I,J, firstK, well.get_lgr_well_tag().value_or(""));
+                auto upperIJK = grid.getLGR_fatherIJK(I,J, lastK, well.get_lgr_well_tag().value_or(""));
+
+                iWell[Ix::IHead] = lowerIJK[0] + 1;
+                iWell[Ix::JHead] = lowerIJK[1] + 1;
+                iWell[Ix::FirstK] = lowerIJK[2] + 1;
+                iWell[Ix::LastK]  = upperIJK[2] + 1;
+            }
+        }
+
+        template <class IWellArray>
+        void staticContrib(const Opm::Well&                well,
+                           const Opm::GasLiftOpt&          glo,
+                           const Opm::WellTestConfig&      wtest_config,
+                           const Opm::WellTestState&       wtest_state,
+                           const Opm::SummaryState&        st,
+                           const std::size_t               msWellID,
+                           const std::map <const std::string, size_t>&  GroupMapNameInd,
+                           IWellArray&                     iWell,
+                           std::optional<std::reference_wrapper<const Opm::EclipseGrid>>
+                                                           grid = std::nullopt,
+                           bool global_grid = true)
+        {
+            using Ix = VI::IWell::index;
+            iWell[Ix::Status] = wellStatus(well.getStatus());
+            // Global Well - NON LGR
+            if (!well.is_lgr_well()){
+                staticContribWellHead(well, iWell);
+            }
+            // LGR Well
+            else {
+                // If well is in the global grid
+                if (global_grid) {
+                    staticContribWellHeadLGR(well, iWell, grid.value().get());
```

Comment:
Does the calling code guarantee that `grid` has a value here?  If not, you probably need to check `grid.has_value()` too.